### PR TITLE
Feature/discovery server/data structures

### PR DESCRIPTION
**Pull request by jparisu**
All interface DDB to PDPServer2
new directory discovery/database
change in CMakeList to not force c++11

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2655/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/67/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2468/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2665/)

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return cache_change_;
+}
+
+
+
+
+} /* namespace ddb */
+} /* namespace dds */
+} /* namespace fastdds */
+} /* namespace eprosima */
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.cpp:38]: Please, add blank line at the end of the file

Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataQueueInfo.cpp:38]: In general, please add a blank line to the end of the file

**Review by EduPonz — CHANGES_REQUESTED** [review]
Some minor things

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+        bool pending()
+        {
+            return pending_;
+        }
+
+    private:
+
+        DiscoveryDataBase* db_;
+        eprosima::fastrtps::rtps::CacheChange_t* cache_;
```
Comment (EduPonz) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp:90]: I think `change` is clearer. Update in cpp.
```suggestion
        eprosima::fastrtps::rtps::CacheChange_t* change_;
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM. Please uncrustify

**Pull request merged**